### PR TITLE
[BEAM-1928] Add PCollections Utilities

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Convert a PCollection into a Proto

### CODE REVIEW

**Comment by tgroh** [comment]
R: @lukecwik

**Comment by tgroh** [comment]
Rebased on top of #2517.

**Comment by tgroh** [comment]
retest this please

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollections.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  private static IsBounded fromProto(RunnerApi.IsBounded isBounded) {
+    switch (isBounded) {
+      case BOUNDED:
+        return IsBounded.BOUNDED;
+      case UNBOUNDED:
+        return IsBounded.UNBOUNDED;
+      case UNRECOGNIZED:
+      default:
+        // Whether or not it is proto that cannot recognize it (due to the version of the
```
Comment (lukecwik) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollections.java]: Whether or not it is proto that cannot recognize it  -> Whether or this enum cannot be recognized

File: runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/SdkComponentsTest.java [hunk 2]
```diff
@@ -112,14 +112,15 @@ public void registerTransformIdEmptyFullName() {
...
+  public void registerPCollection() throws IOException {
     PCollection<Long> pCollection = pipeline.apply(CountingInput.unbounded()).setName("foo");
     String id = components.registerPCollection(pCollection);
     assertThat(id, equalTo("foo"));
+    components.toComponents().getPcollectionsOrThrow(id);
   }
 
   @Test
-  public void registerPCollectionExistingNameCollision() {
+  public void registerPCollectionExistingNameCollision()  throws IOException {
```
Comment (lukecwik) [inline runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/SdkComponentsTest.java]: nit: remove extra space between () and throws

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollections.java (hunk 1 above)
Comment (tgroh) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PCollections.java]: Done.

File: runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/SdkComponentsTest.java (hunk 2 above)
Comment (tgroh) [inline runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/SdkComponentsTest.java]: Done.

**Comment by lukecwik** [comment]
LGTM

**Pull request closed without merging**